archunit/src/main/java/com/tngtech/archunit/core/domain/AccessTarget.java

```diff
@@ -344,9 +343,9 @@ public JavaClass getRawReturnType() {
         @Override
         @PublicAPI(usage = ACCESS)
         public ThrowsClause<CodeUnitCallTarget> getThrowsClause() {
-            List<ThrowsClause<JavaCodeUnit>> resolvedThrowsClauses = FluentIterable.from(resolve())
-                    .transform(toGuava(JavaCodeUnit.Functions.Get.throwsClause()))
-                    .toList();
+            List<ThrowsClause<JavaCodeUnit>> resolvedThrowsClauses = resolve().stream()
```

Comment:
It seems a little random to just replace this one usage of `FluentIterable` :thinking: Maybe drop this from the commit and we have some commit/PR later that replaces all `FluentIterable` by Java Streams?

Reply:
I agree it seems random, but this one gave me a compilation error locally; When I revert it I once again get:
```
> Task :archunit:compileJava FAILED
/home/tim/Documents/workspace/archunit/archunit/src/main/java/com/tngtech/archunit/core/domain/AccessTarget.java:349: error: incompatible types: ImmutableList<ThrowsClause<CAP#1>> cannot be converted to List<ThrowsClause<JavaCodeUnit>>
                    .toList();
                           ^
  where CAP#1 is a fresh type-variable:
    CAP#1 extends JavaCodeUnit from capture of ? extends JavaCodeUnit
1 error
```

That's running on JDK 17-open as installed through sdkman:
```
openjdk version "17" 2021-09-14
OpenJDK Runtime Environment (build 17+35-2724)
OpenJDK 64-Bit Server VM (build 17+35-2724, mixed mode, sharing)
```

I had tried to change the assignment, but that failed for me as well; figured a first move away from Guava wouldn't hurt on Java 8+.

Do you see any other options to resolve this issue?

Reply:
Ah, okay, that seems odd :joy: But no, then it makes sense to have it also in the JDK 17 commit, because it's directly related :+1: Would just maybe also add it as a detail to the commit message that this one `FluentIterable` had to be replaced, since it doesn't compile anymore with JDK 17. So it is clear when someone looks at the commit later on why only this was picked to be replaced :slightly_smiling_face:

Reply:
Pushed it now as a separate commit; If on the other hand we decide to first & separately drop Java 7, it would also be possible to convert more/all FluentIterables to Streams before adding JDK 17.